test/map/map.cpp

```diff
@@ -228,3 +228,39 @@ TEST(Map, RemoveLayer) {
 
     test::checkImage("test/fixtures/map/remove_layer", test::render(map));
 }
+
+TEST(Map, Classes) {
+    MapTest test;
+
+    Map map(test.view, test.fileSource, MapMode::Still);
+    map.setStyleJSON(util::read_file("test/fixtures/api/empty.json"));
+
+    EXPECT_EQ(map.getTransitionOptions().duration->count(), 0);
+
+    auto duration = mbgl::Milliseconds(300);
+    map.setTransitionOptions({ { duration } });
+    EXPECT_EQ(map.getTransitionOptions().duration->count(),
```

Comment:
Yes and no - with the current implementation I get `invalid operands to binary expression ('const std::experimental::fundamentals_v1::optional<std::__1::chrono::duration<long long, std::__1::ratio<1, 1000000000> > >' and 'const std::__1::chrono::duration<long long, std::__1::ratio<1, 1000> >')` - but if I encapsulate `duration` inside `mbgl::Duration` then it works fine 👍 Thanks!